### PR TITLE
[PDB-1987] Check box filters with no default value set, display the first checkbox as selected

### PR DESCRIPTION
**Pull request by denisprotopopov**
- renamed the component property that controls first value selection, from `defaultIfEmpty` to `useFirstValue`, to be consistent with dashboards and common-ui plugins across the platform

### CODE REVIEW

**Comment by denisprotopopov** [comment]
should be merge together:
https://github.com/webdetails/cdf/pull/974
https://github.com/webdetails/cde/pull/805
https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1224
https://github.com/pentaho/pentaho-platform-plugin-dashboards/pull/436

**Comment by denisprotopopov** [comment]
@afrjorge @pamval please review the changes

File: core-js/src/main/javascript/cdf/components/MultiButtonComponent.js [hunk 1]
```diff
@@ -143,7 +143,7 @@ define([
           if(!myself.isMultiple) { break; }
         }
       }
-      if(((!foundDefault && !myself.isMultiple) || (!foundDefault && myself.isMultiple && myself.defaultIfEmpty))
+      if(((!foundDefault && !myself.isMultiple && myself.useFirstValue) || (!foundDefault && myself.isMultiple && myself.useFirstValue))
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf/components/MultiButtonComponent.js]: Why did you change the first part to `(!foundDefault && !myself.isMultiple && myself.useFirstValue)`?
The idea is to simply replace the `defaultIfEmpty` with `useFirstValue`, this extra validation breaks the unit tests.

File: core-js/src/main/javascript/cdf/components/ToggleButtonBaseComponent.js [hunk 2]
```diff
@@ -71,12 +71,12 @@ define(['../lib/jquery', './InputBaseComponent'], function($, InputBaseComponent
         }
 
         if(myself.type == 'radio' || myself.type.toLowerCase() == 'radiocomponent') {
-          if((i == 0 && !hasCurrentVal) || (hasCurrentVal && (myArray[i][vid] == currentVal))) {
+          if((i == 0 && !hasCurrentVal && myself.useFirstValue) || (hasCurrentVal && (myArray[i][vid] == currentVal))) {
```
Comment (afrjorge) [inline core-js/src/main/javascript/cdf/components/ToggleButtonBaseComponent.js]: Same as above.

File: core-js/src/main/javascript/cdf/components/MultiButtonComponent.js (hunk 1 above)
Comment (denisprotopopov) [reply core-js/src/main/javascript/cdf/components/MultiButtonComponent.js]: because without this changes first value selected
